pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/OpChainExecutionContext.java

```diff
@@ -123,4 +126,8 @@ public ServerPlanRequestContext getLeafStageContext() {
   public void setLeafStageContext(ServerPlanRequestContext leafStageContext) {
     _leafStageContext = leafStageContext;
   }
+
+  public ThreadExecutionContext getParentContext() {
```

Comment:
This method and the attribute should also be nullable.

Reply:
Done.